### PR TITLE
DOCSUP-8661: MySQL dictionary source: the fail_on_connection_loss config parameter.

### PR DESCRIPTION
**Pull request by sevirov**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category:

- Documentation for #21237

### CODE REVIEW

File: docs/en/sql-reference/dictionaries/external-dictionaries/external-dicts-dict-sources.md [hunk 1]
```diff
@@ -523,6 +525,8 @@ Setting fields:
 
 -   `invalidate_query` – Query for checking the dictionary status. Optional parameter. Read more in the section [Updating dictionaries](../../../sql-reference/dictionaries/external-dictionaries/external-dicts-dict-lifetime.md).
 
+-   `fail_on_connection_loss` – The configuration parameter that controls unexpected connection loss during query execution. By default, false. ClickHouse connects to MySQL servers using secure protocols ([SSL](../../../operations/server-configuration-parameters/settings.md#server_configuration_parameters-openssl)/[TLS](https://en.wikipedia.org/wiki/Transport_Layer_Security)). Occasionally errors might occur in the transport communication between client and server, in which case the server would abort the client connection. Note that retrying leads to increased response times. Retries can be avoided by setting the parameter to true.
```
Comment (kssenii) [inline docs/en/sql-reference/dictionaries/external-dictionaries/external-dicts-dict-sources.md:528]: It is better to make it concise, seems like too much text and no actual info. The main point: If `fail_on_connection_loss` is `true`, then there will be an exception about connection loss straight away if there was no connection when query was executed. If  `fail_on_connection_loss` is `false`, then there will be attempts to retry queries `n` times. where `n` is not configurable now and default (`n = 3`) is always used.

Comment (sevirov) [reply docs/en/sql-reference/dictionaries/external-dictionaries/external-dicts-dict-sources.md:528]: > It is better to make it concise, seems like too much text and no actual info. The main point: If `fail_on_connection_loss` is `true`, then there will be an exception about connection loss straight away if there was no connection when query was executed. If `fail_on_connection_loss` is `false`, then there will be attempts to retry queries `n` times. where `n` is not configurable now and default (`n = 3`) is always used.

Ok, thank you. I will fix it.

File: docs/en/sql-reference/dictionaries/external-dictionaries/external-dicts-dict-sources.md [hunk 2]
```diff
@@ -523,6 +525,8 @@ Setting fields:
 
 -   `invalidate_query` – Query for checking the dictionary status. Optional parameter. Read more in the section [Updating dictionaries](../../../sql-reference/dictionaries/external-dictionaries/external-dicts-dict-lifetime.md).
 
+-   `fail_on_connection_loss` – The configuration parameter that controls unexpected connection loss during query execution. If `true`, then there will be an exception about connection loss straight away if there was no connection when the query was executed. If `false`, then there will be attempts to retry queries three times. By default, `false`.
```
Comment (olgarev) [inline docs/en/sql-reference/dictionaries/external-dictionaries/external-dicts-dict-sources.md:528]:
```suggestion
-   `fail_on_connection_loss` – The configuration parameter that controls unexpected connection loss during query execution. If `true`, an exception is thrown if the connection between client and server was lost while the query was executed. If `false`, the ClickHouse server retries to execute the query three times. Default value: `false`.
```

File: docs/ru/sql-reference/dictionaries/external-dictionaries/external-dicts-dict-sources.md [hunk 3]
```diff
@@ -492,6 +494,8 @@ SOURCE(MYSQL(
 
 -   `invalidate_query` — запрос для проверки статуса словаря. Необязательный параметр. Читайте подробнее в разделе [Обновление словарей](external-dicts-dict-lifetime.md).
 
+-   `fail_on_connection_loss` – Параметр конфигурации, контролирующий неожиданную потерю соединения во время выполнения запроса. Если `true`, то сразу же возникнет исключение о том, что при выполнении запроса соединения не было. Если `false`, то будут попытки повторить запросы три раза. По умолчанию `false`.
```
Comment (olgarev) [inline docs/ru/sql-reference/dictionaries/external-dictionaries/external-dicts-dict-sources.md:497]:
```suggestion
-   `fail_on_connection_loss` – параметр конфигурации, контролирующий поведение сервера при потере соединения во время выполнения запроса. Если значение `true`, то при потере соединения во время выполнения запроса исключение генерируется сразу же. Если значение `false`, то сервер повторно попытается выполнить запрос три раза прежде чем сгенерировать исключение. Значение по умолчанию: `false`.
```

**Comment by sevirov** [comment]
@kssenii Please, review this PR.

File: docs/en/sql-reference/dictionaries/external-dictionaries/external-dicts-dict-sources.md [hunk 4]
```diff
@@ -523,6 +525,8 @@ Setting fields:
 
 -   `invalidate_query` – Query for checking the dictionary status. Optional parameter. Read more in the section [Updating dictionaries](../../../sql-reference/dictionaries/external-dictionaries/external-dicts-dict-lifetime.md).
 
+-   `fail_on_connection_loss` – The configuration parameter that controls unexpected connection loss during query execution. If `true`, an exception is thrown immediately if the connection between client and server was lost while the query was executed. If `false`, the ClickHouse server retries to execute the query three times before throwing an exception. Note that retrying leads to increased response times. Default value: `false`.
```
Comment (kssenii) [inline docs/en/sql-reference/dictionaries/external-dictionaries/external-dicts-dict-sources.md:528]: > if the connection between client and server was lost while the query was executed

Actually not `while`. It is checked only before the query is executed.

**Review by kssenii — APPROVED** [review]

**Pull request merged**